includes/class-wc-gzd-customer-helper.php

```diff
@@ -430,6 +430,13 @@ public function account_cleanup() {
 		if ( ! empty( $user_query->results ) ) {
 			foreach ( $user_query->results as $user ) {
 
+				// Don't delete users with additional/multiple roles (like `editor` or `shop_manager`)
+				foreach ($roles as $role) {
+					if (!array_key_exists($role, $user->roles) || !$user->roles[$role]) {
+						return;
+					}
+				}
```

Comment:
Ich habe mich gegen die `'role__not_in' => ['administrator', 'shop_manager'],`-Variante entschieden, weil es beliebig viel Rollen geben kann, welche nie gelöscht werden sollten.

Stattdessen überprüfe ich, ob der User ausschließlich die Rollen von `$this->get_double_opt_in_user_roles()` hat. Das ist in diesem Fall aktuell nur die `customer`-Rolle.

Reply:
Ok. Ich denke das passt aber noch nicht so ganz. Es kann ja durchaus sein, dass ein User z.B. nur eine der verfügbaren Rollen `$roles` für den DOI besitzt (wenn z.B. via Filter weitere Rollen explizit registriert wurden). Das würde in diesem Fall dazu führen, dass der User nicht gelöscht wird. Ich finde die Variante in der Query eigentlich sinnvoller.

Reply:
Stimmt, das habe ich nicht ganz durchüberlegt. Hab den PR angegpasst.